src/game_functions/board_functions.c

```diff
@@ -0,0 +1,90 @@
+#include "board_functions.h"
+#include <stdbool.h>
+#include <stdlib.h>
+
+
+struct positionFromIndex calcPositionFromIndex (int index, int width) {
+    struct positionFromIndex return_val;
+    return_val.x = index / width;
+    return_val.y = index % width;
```

Comment:
This makes the assumption that the board is square. Although we are only working with a square, if we ever in the future are required to use this function for a board that is not square, (we probably will not but whatever) then it may be important to use the height instead of the width for the second value. Up to you to change it or not.

Reply:
No, this doesn't make that assumption. To calculate the `y` value of a given index, you just need the columns, not the rows. The assumption made is that every row will have the same number of columns (which is a valid assumption to make). I'll explain the process on discord.

Reply:
oh sorry I meant that it did not work for uneven grid types. If you think that it is a valid assumption to make then that is fine.